security-proxy/src/main/java/org/georchestra/security/Proxy.java

```diff
@@ -787,7 +738,58 @@ private Header extractHeaderSetCookie(HttpResponse proxiedResponse) {
 
     @VisibleForTesting
     protected HttpResponse executeHttpRequest(CloseableHttpAsyncClient httpclient, HttpRequestBase proxyingRequest) throws IOException, TimeoutException, ExecutionException, InterruptedException {
-        Future<HttpResponse> future = httpclient.execute(proxyingRequest, null);
+        CompletableFuture<HttpResponse> future = new CompletableFuture<HttpResponse>();
+
+        PipedOutputStream pos = new PipedOutputStream();
+        PipedInputStream pis = new PipedInputStream(pos);
+        WritableByteChannel channel = Channels.newChannel(pos);
+
+        HttpAsyncResponseConsumer<Boolean> consumer = new AbstractAsyncResponseConsumer<Boolean>() {
+
+            private ByteBuffer bbuf =  ByteBuffer.allocate(8192);
+            private HttpResponse httpResponse;
+
+            protected void onEntityEnclosed(HttpEntity entity, ContentType contentType) throws IOException {
+                HttpEntity streamEntity = new InputStreamEntity(pis, contentType);
+                httpResponse.setEntity(streamEntity);
+                future.complete(httpResponse);
+            }
+
+            protected void onContentReceived(ContentDecoder decoder, IOControl ioctrl) throws IOException {
+                int bytesRead = decoder.read(this.bbuf);
+                if (bytesRead > 0) {
+                    this.bbuf.flip();
+                    channel.write(this.bbuf);
```

Comment:
Beware the output channel have written less than bbuf.remaining bytes, I guess we should make it something like ```while(this.bbuf.hasRemaining()) {
                    	channel.write(this.bbuf);
                    }```

Reply:
thank you !